### PR TITLE
[BD-03] [TNL-7742] [BB-3370] Add LTI 1.1 config on model

### PR DESCRIPTION
**Pull request by xitij2000**
Add support for storing LTI1.1 configuration in the database.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [BLENDED-688](https://openedx.atlassian.net/browse/BLENDED-688) to keep track of it in Jira.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: lti_consumer/models.py
Comment (stvstnfrd) [inline lti_consumer/models.py:76]: What is the primary ID now?

File: lti_consumer/migrations/0006_add_on_model_config_for_lti_1p1.py
Comment (stvstnfrd) [inline lti_consumer/migrations/0006_add_on_model_config_for_lti_1p1.py:44]: Probably not a big deal, but I wonder if the default should now be `CONFIG_ON_DB`..

**Review by stvstnfrd — COMMENTED** [review]
I don't know LTI particularly well, but this seems reasonable from my understanding of what we need on the discussions side.

I defer to someone that knows LTI better, though.

File: lti_consumer/models.py
Comment (xitij2000) [reply lti_consumer/models.py:76]: The original default auto-generated ID is till the primary ID.

File: lti_consumer/migrations/0006_add_on_model_config_for_lti_1p1.py
Comment (xitij2000) [reply lti_consumer/migrations/0006_add_on_model_config_for_lti_1p1.py:44]: The config on db path isn't fully developed, and doesn't support LTI1.3, so until all the paths for that are supported I think we should default to config on XBlock.

**Comment by xitij2000** [comment]
@stvstnfrd I upgraded the dependencies to pull in the latest version of edx-lint and that raised a lot of quality warnings. I tried to fix the dependencies but it was a bit fragile so I decided to instead use that time fixiing the quality issues. Those are in a separate commit to make it easier to review the actual content of the PR.

File: lti_consumer/tests/unit/test_models.py
Comment (shimulch) [inline lti_consumer/tests/unit/test_models.py:119]: You need to update LTI configuration id to ``3`` for the LTI AGS test [here](https://github.com/open-craft/xblock-lti-consumer/blob/kshitij/lti-on-model-config-lti1.1/lti_consumer/tests/unit/test_models.py#L129). Because your change on this PR creates a new LTI configuration and ``lti_1p3_config`` is the last one. I think this is why the test is failing.

**Review by shimulch — CHANGES_REQUESTED** [review]

File: lti_consumer/tests/unit/test_models.py
Comment (xitij2000) [reply lti_consumer/tests/unit/test_models.py]: Done.

**Review by shimulch — APPROVED** [review]
LGTM :+1:

- [x] I tested this on the sandbox attached with https://github.com/edx/edx-platform/pull/25910. That PR has a dependency on this PR. I've created LTI 1.1 configuration from Django Admin. It works fine.
- [x] I read through the code
- [N/A] I checked for accessibility issues
- [x] Includes documentation
- [N/A] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Review by stvstnfrd — APPROVED** [review]
LGTM

**Comment by sarina** [comment]
My understanding is this needs a rebase, then we can merge?

**Comment by stvstnfrd** [comment]
@xitij2000 Looks like this just needs squashed/rebased and then is good to go?
Let me know if you/open-craft don't have the merge bit here, and I can hit the button :+1:

**Comment by xitij2000** [comment]
@stvstnfrd I had already rebased it into 2 commits, I've now brought it down to 1. 

I don't think anyone at OC has merge rights here, so you'll need to do the merging.

**Pull request merged**
